packages/tx/src/transaction.ts

```diff
@@ -21,290 +21,255 @@ const N_DIV_2 = new BN('7fffffffffffffffffffffffffffffff5d576e7357a4501ddfe92f46
  * An Ethereum transaction.
  */
 export default class Transaction {
-  public raw!: Buffer[]
-  public nonce!: Buffer
-  public gasLimit!: Buffer
-  public gasPrice!: Buffer
-  public to!: Buffer
-  public value!: Buffer
-  public data!: Buffer
-  public v!: Buffer
-  public r!: Buffer
-  public s!: Buffer
-
-  private _common: Common
-  private _senderPubKey?: Buffer
-  protected _from?: Buffer
+  public readonly common: Common
+  public readonly nonce: BN
+  public readonly gasLimit: BN
+  public readonly gasPrice: BN
+  public readonly to?: Address
+  public readonly value: BN
+  public readonly data: Buffer
+  public readonly v?: BN
+  public readonly r?: BN
+  public readonly s?: BN
+
+  public static fromTxData(txData: TxData, common?: Common) {
+    const { nonce, gasLimit, gasPrice, to, value, data, v, r, s } = txData
+
+    return new Transaction(
+      common,
+      new BN(toBuffer(nonce || '0x')),
+      new BN(toBuffer(gasPrice || '0x')),
+      new BN(toBuffer(gasLimit || '0x')),
+      to ? new Address(toBuffer(to)) : undefined,
+      new BN(toBuffer(value || '0x')),
+      toBuffer(data || '0x'),
+      new BN(toBuffer(v || '0x')),
+      new BN(toBuffer(r || '0x')),
+      new BN(toBuffer(s || '0x')),
+    )
+  }
+
+  public static fromRlpSerializedTx(serialized: Buffer, common?: Common) {
+    const values = rlp.decode(serialized)
+
+    if (!Array.isArray(values)) {
+      throw new Error('Invalid serialized tx input. Must be array')
+    }
+
+    return this.fromValuesArray(values, common)
+  }
+
+  public static fromValuesArray(values: Buffer[], common?: Common) {
+    if (values.length > 9) {
+      throw new Error('Invalid transaction. More values than expected were received')
+    }
+
+    const [nonce, gasPrice, gasLimit, to, value, data, v, r, s] = values
```

Comment:
What if the length is `< 9`? 

Maybe `9` and `6` should be the only valid lengths. Like signed and unsigned txs.

Reply:
good idea, added in https://github.com/ethereumjs/ethereumjs-vm/pull/812/commits/d6cc6bfb89275bdfb5ac498db0960ff12cd23b52